CMU-Robotics-Club/RoboBuggy2
# Auton overtake feature, as well as pylint fixes

## What type of PR is this? (check all applicable)


## Description

- Added path planner.
- Refactored autonsystem to run path planning.
- Extended trajectory.py, world.py with methods required by path planner.
- Added the ability to initialize new trajectory from a numpy array.
- Fixed bug in path projection and vectorized the code.
- Added command line options to autonsystem and integrated these options into the sim launch files.
- Modified topic names so that NAND/SC topics do not conflict.
- Added half-plane constraints to MPC code (this is not used for now, but might be interfaced with path planning later)
- Adjusted Stanley controller heading error weight to reduce oscillation
- Cleaned up code to conform to pylint

## Related Tickets & Documents



- Related Issue #
- Closes #

## QA Instructions, Screenshots, Recordings

To test auton overtaking in sim: run sim_2d_2buggies.launch
You will be able to visualize the movements of the 2 simulated buggies in Foxglove.

What you should see on the map visualization are 2 buggies. One will move faster and overtake another, staying 2 meters away at the closest.

Here's the ROSbag of the demo run: https://drive.google.com/file/d/1913GAiZg_PoAF8P0pv_pyOv-dJqz8pLd/view?usp=drive_link

## Added/updated tests?
_We encourage you to keep the code coverage percentage at 80% and above._

      have not been included_
